src/operator/crop-inl.h

```diff
@@ -0,0 +1,181 @@
+/*!
+ * Copyright (c) 2015 by Contributors
+ * \file crop-inl.h
+ * \brief
+ * \author Wei Wu
+*/
+#ifndef MXNET_OPERATOR_CROP_INL_H_
+#define MXNET_OPERATOR_CROP_INL_H_
+#include <dmlc/logging.h>
+#include <dmlc/parameter.h>
+#include <mxnet/operator.h>
+#include <cstring>
+#include <map>
+#include <string>
+#include <vector>
+#include <utility>
+#include "./operator_common.h"
+
+namespace mxnet {
+namespace op {
+
+namespace crop_enum {
+enum CropOpInputs {kData, kCropLike};
+enum CropOpOutputs {kOut};
+}  // namespace crop_enum
+
+struct CropParam : public dmlc::Parameter<CropParam> {
+  TShape offset;
+  bool center_crop;
+  DMLC_DECLARE_PARAMETER(CropParam) {
+    int shape[] = {0, 0};
+    DMLC_DECLARE_FIELD(offset).set_default(TShape(shape, shape + 2))
+    .describe("corp offset coordinate: (y, x)");
+    DMLC_DECLARE_FIELD(center_crop).set_default(false)

```

Comment:
yes, it is a bit hard for explicitly the size, the reason is that we cannot konw the crop size before we do the crop operator, because it depends on the input image size(fcn-32s) and the intermdiate layer size(fcn-16s and fcn-8s), and in caffe, https://github.com/BVLC/caffe/pull/1976 it add more code for the crop layer.
the crop layer is used for algiment between two layer, like in spp-net, the proposal box between the input layer and conv5 feature map, and in spp-net, it used a special crop like layer doing the algiment.
the crop layer in caffe is also need the auto alignment in https://github.com/longjon/caffe/blob/future/src/caffe/layers/crop_layer.cpp#L13

ps: in the future, we should add the value of crop in the symbol side, instead of  hand calculation in this example.